simulations/src/node/carnot/mod.rs

```diff
@@ -381,8 +381,11 @@ impl<L: UpdateableLeaderSelection, O: Overlay<LeaderSelection = L>> CarnotNode<O
                     parent_block=%block.parent(),
                     "receive approve message"
                 );
-                let (new, out) = self.engine.approve_block(block);
+                let block_grandparent_view = block.view - View(3);
```

Comment:
it might be more if we have failure rounds?

Reply:
Yeah, I was thinking about that. In that case we would have to go keep blocks till the last healthy one 🤔